akamai/cache-client_test.go

```diff
@@ -129,19 +143,54 @@ func (as *akamaiServer) akamaiHandler(w http.ResponseWriter, r *http.Request) {
 	as.sendResponse(w, resp)
 }
 
+func (as *akamaiServer) checkSignature(r *http.Request, body []byte) error {
+	bodyHash := sha256.Sum256(body)
+	bodyHashB64 := base64.StdEncoding.EncodeToString(bodyHash[:])
+
+	authorization := r.Header.Get("Authorization")
+	authValues := make(map[string]string)
+	for _, v := range strings.Split(authorization, ";") {
+		splitValue := strings.Split(v, "=")
+		authValues[splitValue[0]] = splitValue[1]
+	}
+	headerTimestamp := authValues["timestamp"]
+	shortenedHeader := strings.Split(authorization, "signature=")[0]
+	// The signature is just long enough to get b64 padding which gets messed up
+	// with splitting on "=", so don't use the map to extract it.
+	signature := strings.Split(authorization, "signature=")[1]
+	hostPort := strings.Split(as.URL, "://")[1]
```

Comment:
Why not `r.URL.Host` here?

Reply:
`httptest.URL` is a string. :-)

Reply:
So is `r.URL.Host`?

Reply:
Haha oops I didn't see the `r` in there. I guess that does work fine - and would let me unroll the diffs that made `akamaiServer` embed `httptest.Server`

Reply:
Upon testing, it appears that `r.URL.Host` is empty in the server-handling side of things. So I don't think I can do this.

Reply:
Ah bummer, that is annoying.